docs/model_developers/unmanaged_resources/basic_example_handler.py

```diff
@@ -0,0 +1,61 @@
+import re
+from collections import abc
+
+import pydantic
+
+from inmanta import resources
+from inmanta.data.model import ResourceIdStr
+from inmanta.agent.handler import provider, DiscoveryHandler, HandlerContext
+from inmanta.resources import resource, DiscoveryResource
+
+
+@resource("my_module::InterfaceDiscovery", agent="host.name", id_attribute="host")
+class InterfaceDiscovery(DiscoveryResource):
+    fields = ("host", "name_filter")
+
+    host: str
+    name_filter: str
+
+    @staticmethod
+    def get_host(exporter, resource):
+        return resource.host.name
+
+
+class UnmanagedInterface(pydantic.BaseModel):
+    """
+    Datastructure used by the InterfaceDiscoveryHandler to return the attributes
+    of its discovered resources.
+    """
+    host: str
+    interface_name: str
+    ip_address: pydantic.IPvAnyAddress
+
+
+@provider("my_module::InterfaceDiscovery", name="interface_discovery_handler")
+class InterfaceDiscoveryHandler(DiscoveryHandler[InterfaceDiscovery, UnmanagedInterface]):
+    def discover_resources(
+        self, ctx: HandlerContext, discovery_resource: InterfaceDiscovery
+    ) -> abc.Mapping[ResourceIdStr, UnmanagedInterface]:
```

Comment:
```suggestion
    ) -> dict[ResourceIdStr, UnmanagedInterface]:
```

Not very important to me, but ideally I think concrete method implementation should have abstract argument types and concrete return types, unless flexibility is required for future changes. In the ABC I like the `abc.Mapping` return type because it allows children some flexibility.